designs/2024-support-l10n-rule-messages/README.md

```diff
@@ -0,0 +1,185 @@
+- Repo: eslint/eslint
+- Start Date: 2024-12-09
+- RFC PR: (leave this empty, to be filled in later)
+- Authors: [gweesin](https://github.com/gweesin)
+
+# Support Localization of Rule Messages
+
+## Summary
+
+<!-- One-paragraph explanation of the feature. -->
+
+This RFC proposes adding support for localization of rule messages in ESLint. The goal is to allow users to provide translations for rule messages, enabling ESLint to display messages in different languages based on user preferences. This feature aims to improve the accessibility and usability of ESLint for non-English speaking users.
+
+## Motivation
+
+<!-- Why are we doing this? What use cases does it support? What is the expected
+outcome? -->
+
+The primary motivation for this feature is to make ESLint more accessible to a global audience. Currently, ESLint rule messages are only available in English, which can be a barrier for non-English speaking users. By allowing rule messages to be localized, we can improve the user experience for developers who prefer to work in their native language. This change will help ESLint reach a broader audience and make it easier for developers around the world to understand and fix linting issues in their code.
+
+## Detailed Design
+
+<!--
+   This is the bulk of the RFC.
+
+   Explain the design with enough detail that someone familiar with ESLint
+   can implement it by reading this document. Please get into specifics
+   of your approach, corner cases, and examples of how the change will be
+   used. Be sure to define any new terms in this section.
+-->
+
+To implement localization of rule messages in ESLint, we will support two ways to define the localization information:
+
+1. Support for plugin definition messages with locale files.
+   plugins can define localized messages directly within their rule definitions so that all the messages can be supported in the same repository. This approach allows plugin authors to provide translations for their rule messages without relying on external files.
+
+   Example:
+
+   ```js
+   module.exports = {
+      meta: {
+         messages: {
+            someMessageId: {
+               en: "Default message in English",
+               es: "Mensaje predeterminado en español"
+            }
+         }
+      },
+      create(context) {
+         return {
+            Identifier(node) {
+               context.report({
+                  node,
+                  messageId: "someMessageId"
+               });
+            }
+         };
+      }
+   };
+   ```
+
+2. Support a way using `eslint.config.js` file to configure message info by message ID.
+   This way should allow users to provide some plugins to update other plugins' messages. This approach allows users to customize the messages for rules provided by plugins.
+
+   Example:
+    
+   ```js
+   module.exports = {
+     locale: "es",
```

Comment:
What does `locale` do here?

Does it only apply to files matching `files`/`ignores`? (This is how config objects work in the config array, there's no such thing as a global setting.)

Reply:
Just define the expected language the user wants to see in the lint error messages.

Reply:
Sorry, I think I was unclear. If you have a config like this:

```js
{
    files: ["**/*.js"],
    locale: "es"
}
```

That config is only applied when the file being linted matches `**/*.js`. Is that was you intended?